src/storage-controller/src/lib.rs

```diff
@@ -1129,7 +1126,85 @@ where
                 .expect("poisoned")
                 .insert(id, statistics::StatsInitState(BTreeMap::new()));
 
-            client.send(StorageCommand::CreateSinks(vec![cmd]));
+            client.send(StorageCommand::RunSinks(vec![cmd]));
+        }
+        Ok(())
+    }
+
+    /// Create the sinks described by the `ExportDescription`.
+    async fn update_export_connection(
+        &mut self,
+        exports: BTreeMap<GlobalId, StorageSinkConnection>,
+    ) -> Result<(), StorageError> {
+        for (id, connection) in exports {
+            let export = self.export_mut(id).expect("export exists");
+            let current_sink = export.description.sink.clone();
+            export.description.sink.connection = connection;
+
+            // Ensure compatibility
+            current_sink.alter_compatible(id, &export.description.sink)?;
+
+            // Get export data
+            let value = MetadataExportFetcher::get_stash_collection()
+                .peek_key_one(&mut self.stash, id.into_proto())
+                .await?
+                .expect("known to exist");
+
+            let mut durable_export_data = DurableExportMetadata::from_proto(value)
+                .map_err(|e| StorageError::IOError(e.into()))?;
+
+            // Get updated export state
+            let export = self.export(id).expect("export exists").clone();
+
+            let from_collection = self.collection(export.description.sink.from)?;
+            let from_storage_metadata = from_collection.collection_metadata.clone();
+
+            let read_capability = export.read_capability.to_owned();
+
+            // Downgrade this since--this mostly informs us whether or not we
+            // want to read the snapshot.
+            durable_export_data
+                .initial_as_of
+                .downgrade(&read_capability);
```

Comment:
- [ ] i think there should be some more explanation here; is it ensuring we don't re-read the snapshot? do we actually need to do this? the sink itself should have upper recorded in the progress topic right?

Reply:
The docstring and implementation of downgrade here is meaningful to look at w/r/t what happens here––added a nudge to check that out to this comment. fwiw if we received the alter connection before we read anything in the sink, we might still want to read the snapshot.